extensions/amp-date-countdown/0.1/amp-date-countdown.js

```diff
@@ -0,0 +1,254 @@
+   * @private
+   */
+  getLocaleWord_(locale) {
+    //https://ctrlq.org/code/19899-google-translate-languages refer to google code
+    const localeWord = {
+      'zh-CN': ['年', '月', '天', '小时', '分钟', '秒'],
+      'zh-TW': ['年', '月', '天', '小時', '分鐘', '秒'],
+      'de': ['Jahren', 'Monaten', 'Tagen', 'Stunden', 'Minuten', 'Sekunden'],
+      'en': ['Years', 'Months', 'Days', 'Hours', 'Minutes', 'Seconds'],
+      'es': ['años', 'meses', 'días', 'horas', 'minutos', 'segundos'],
+      'fr': ['ans', 'mois', 'jours', 'heures', 'minutes', 'secondes'],
+      'id': ['tahun', 'bulan', 'hari', 'jam', 'menit', 'detik'],
+      'it': ['anni', 'mesi', 'giorni', 'ore', 'minuti', 'secondi'],
+      'ja': ['年', 'ヶ月', '日', '時間', '分', '秒'],
+      'ko': ['년', '달', '일', '시간', '분', '초'],
+      'nl': ['jaar', 'maanden', 'dagen', 'uur', 'minuten', 'seconden'],
+      'pt': ['anos', 'meses', 'dias', 'horas', 'minutos', 'segundos'],
+      'ru': ['год', 'месяц', 'день', 'час', 'минута', 'секунда'],
+      'th': ['ปี', 'เดือน', 'วัน', 'ชั่วโมง', 'นาที', 'วินาที'],
+      'tr': ['yıl', 'ay', 'gün', 'saat', 'dakika', 'saniye'],
+      'vi': ['năm', 'tháng', 'ngày', 'giờ', 'phút', 'giây'],
+    };
+    if (localeWord[locale]) {
+      const localeWordList = localeWord[locale];
+      return {
+        'years': localeWordList[0],
+        'months': localeWordList[1],
+        'days': localeWordList[2],
+        'hours': localeWordList[3],
+        'minutes': localeWordList[4],
+        'seconds': localeWordList[5],
+      };
+    } else {
+      user().error(TAG, `Invalid locale '${locale}', return empty locale word`);
+      return {};
+    }
+  }
+
+  /**
+   * @param {number} ms
+   * @return {Object}
+   * @private
+   */
+  getYDHMSFromMs_(ms) {
+
+    const priorityList = {
+      days: 1,
+      hours: 2,
+      minutes: 3,
+      seconds: 4,
+    };
+
+
+    const d = priorityList[this.biggestUnit_] == 1
+      ? (Math.trunc((ms) / (24 * 60 * 60 * 1000))) : 0;
+    const dMs = ms % (24 * 60 * 60 * 1000);
+    const h = priorityList[this.biggestUnit_] == 2
+      ? (Math.trunc((ms) / (60 * 60 * 1000)))
+      : priorityList[this.biggestUnit_] < 2
+        ? (Math.trunc((dMs) / (60 * 60 * 1000)))
+        : 0;
+    const hMs = ms % (60 * 60 * 1000);
+    const m = priorityList[this.biggestUnit_] == 3
+      ? (Math.trunc((ms) / (60 * 1000)))
+      : priorityList[this.biggestUnit_] < 3
+        ? (Math.trunc((hMs) / (60 * 1000)))
+        : 0;
+    const mMs = ms % (60 * 1000);
+    const s = priorityList[this.biggestUnit_] == 4
+      ? (Math.trunc((ms) / (1000))) : (Math.trunc((mMs) / (1000)));
+
+    return {
+      d,
+      dd: this.padStart_(d),
+      h,
+      hh: this.padStart_(h),
+      m,
+      mm: this.padStart_(m),
+      s,
+      ss: this.padStart_(s),
+    };
+  }
+
+  /**
+   * @param {number} input
+   * @return {Object}
+   * @private
+   */
+  padStart_(input) {
+    if (input > 9 || input < 0) {
+      return input.toString();
```

Comment:
A number primitive does not have a `toString` function. You should use the `String()` function instead. e.g.

```js
return String(input);
```